src/api/LiikuntaApolloClient.ts

```diff
@@ -0,0 +1,108 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment, @typescript-eslint/no-explicit-any */
+
+import {
+  ApolloClient,
+  ApolloQueryResult,
+  gql,
+  NormalizedCacheObject,
+  OperationVariables,
+  QueryOptions,
+} from "@apollo/client";
+import { GetStaticPropsContext } from "next";
+
+import { Language, MenuItem } from "../types";
+import mockMenuItems from "./tmp/menuItems";
+
+function getNavigationItems(menuItemsConnection: { nodes: MenuItem[] }) {
+  let menuItems = menuItemsConnection.nodes;
+
+  if (menuItems.length === 0) {
+    menuItems = mockMenuItems;
+  }
+
+  const sortedMenuItems = [...menuItems].sort((a, b) => a.order - b.order);
+  const navigationItems = sortedMenuItems.map(({ order, ...rest }) => rest);
+
+  return navigationItems;
+}
+
+function getSupportedLanguages(
+  languages: Language[],
+  context: GetStaticPropsContext
+) {
+  // NextJS uses locales as is in the slug. The headless CMS has a locale and
+  // a slug field. The slug field is meant to be used in urls. To be able to do
+  // so, we have to provide the slug field value for NextJS when configuring its
+  // i18n module. That's why we are trying to find the slug field from NextJS
+  // locales.
+  const supportedLanguages = languages.filter(({ slug }) =>
+    context.locales.includes(slug)
+  );
+
+  return supportedLanguages;
+}
+
+type WithPageGlobals<T> = T & {
+  global: {
+    language: Language[];
+    menuItems: MenuItem[];
+  };
+};
+
+class LiikuntaApolloClient extends ApolloClient<NormalizedCacheObject> {
+  async pageQuery<T = any, TVariables = OperationVariables>(
```

Comment:
Allows pages to require global data without much duplicate markup. Until `getStaticProps` is available in `_app` this approach allows data that's required on each page to be shared (menu, language and so forth).